src/detours.cpp

```diff
@@ -52,31 +52,94 @@ namespace rhost {
         }
 
         // MessageBoxW
-        decltype(MessageBoxW) *pMessageBoxW = NULL;
+        decltype(MessageBoxW) *pMessageBoxW = nullptr;
         int WINAPI DetourMessageBoxW(HWND hWnd, LPCWSTR lpText, LPCWSTR lpCaption, UINT uType) {
             char ch[1000];
             wcstombs(ch, lpText, _countof(ch));
             return HostMessageBox(ch, uType);
         }
 
         // MessageBoxA
-        decltype(MessageBoxW) *pMessageBoxA = NULL;
+        decltype(MessageBoxA) *pMessageBoxA = nullptr;
         int WINAPI DetourMessageBoxA(HWND hWnd, LPCSTR lpText, LPCSTR lpCaption, UINT uType) {
             return HostMessageBox(lpText, uType);
         }
 
-        void init_ui_detours() {
+        std::unordered_map<HWND, std::string> hWndTitleMap;
+        decltype(CreateWindowExA) *pCreateWindowExA = nullptr;
+        HWND WINAPI DetourCreateWindowExA(DWORD dwExStyle, LPCSTR lpClassName, LPCSTR lpWindowName, DWORD dwStyle, int X, int Y, int nWidth, int nHeight, HWND hWndParent, HMENU hMenu, HINSTANCE hInstance, LPVOID lpParam) {
+            HWND hWnd = pCreateWindowExA(dwExStyle, lpClassName, lpWindowName, dwStyle, X, Y, nWidth, nHeight, hWndParent, hMenu, hInstance, lpParam);
+
+            // A window without parent or a window created to show message
+            size_t len = 0;
+            if (hWnd != nullptr && hWndParent == nullptr && 
+                hWndTitleMap.find(hWnd) == hWndTitleMap.end() && 
+                SUCCEEDED(StringCchLengthA(lpWindowName, STRSAFE_MAX_CCH, &len)) && len > 0) {
+
+                std::string windowTitle = std::string(lpWindowName);
+                std::wstring_convert<std::codecvt_utf8<char>, char> strConverter;
```

Comment:
It if is 'A' then the encoding will be `CP_ACP` (ie console or 'non-Unicode app` code page and not `UTF-8`. Windows doesn't support UTF8 in console.